sequencer/rpc_endpoint/src/rpc/types/codegen.rs

```diff
@@ -679,6 +682,22 @@ impl InvokeTransactionV1 {
 
         serde_json::from_str::<InvokeTransactionV1>(&tx_string).unwrap()
     }
+
+    fn calculate_hash(&self) -> u64 {
+        let mut s = DefaultHasher::new();
+        self.hash(&mut s);
+        s.finish()
+    }
+}
+
+impl Hash for InvokeTransactionV1 {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.max_fee.hash(state);
+        self.signature.hash(state);
+        self.nonce.hash(state);
+        self.sender_address.hash(state);
+        self.calldata.hash(state);
+    }
```

Comment:
I think this might be fine for now but as these structs represent the deserialized RPC types it doesn't make a lot of sense to have the `impl` here. Ideally I think these would be on a util mod on the client side.